src/components/area-plot/AreaPlot.js

```diff
@@ -1,100 +1,183 @@
 import React from 'react';
-import { Area, AreaChart, ResponsiveContainer, Tooltip } from 'recharts';
-import { PLOT_COLORS, PLOT_SECONDARY_COLORS } from '../../constants/plotConstants';
-
+import { Area, AreaChart } from 'recharts';
+import {
+  dataChangeTypes,
+  PLOT_COLORS,
+  PLOT_SECONDARY_COLORS,
+  dataChangeTypeToStackOffsetMapping,
+} from '../../constants/plotConstants';
+import useBrush, { BRUSH_SELECTION_TYPES } from '../../hooks/useBrush';
+import useTooltip from '../../hooks/useTooltip';
+import { getPercent, toPercent } from '../../utils/formatValue';
 import {
+  getAreaPlotDataAnnotationsChangeType,
+  getAreaPlotDataChangeType,
+  getAxisFormat,
   getCategoryAxisDataKey,
   getCategoryValueAxes,
   getData,
   getIsDataPivoted,
   getMargin,
+  getSeriesShowDataAnnotations,
+  getTickFormatterFromDataKey,
   getUniqueValuesOfDataKey,
-  getXAxis,
   getXAxisDataKey,
-  getYAxisDataKey,
+  getYAxis,
+  getYAxisTickFormatter,
+  getCategoryValueAxisByDataKey,
+  getXAxis,
 } from '../../utils/plotConfigGetters';
-import GridLines from '../shared/grid-lines/GridLines';
-import XAxis from '../shared/x-axis/XAxis';
-import YAxis from '../shared/y-axis/YAxis';
-
-import ZenlyticLegend from '../zenlytic-legend/ZenlyticLegend';
+import GeneralChartComponents from '../general-chart-components/GeneralChartComponents';
+import PlotContainer from '../plot-container/PlotContainer';
+import StackedDataAnnotation from './components/stacked-data-annotation/StackedDataAnnotation';
 
-function AreaPlot({ plotConfig = {} }) {
-  const xAxisConfig = getXAxis(plotConfig);
-
-  const xAxisDataKey = getXAxisDataKey(plotConfig);
+function PivotedAreaPlot({ plotConfig }) {
+  const data = getData(plotConfig);
   const categoryAxisDataKey = getCategoryAxisDataKey(plotConfig);
   const uniqueValuesOfCategoryKey = getUniqueValuesOfDataKey(plotConfig, categoryAxisDataKey);
+  const showDataAnnotations = getSeriesShowDataAnnotations(plotConfig);
+  const yAxisTickFormatter = getYAxisTickFormatter(plotConfig);
+  const dataAnnotationsChangeType = getAreaPlotDataAnnotationsChangeType(plotConfig);
+  return uniqueValuesOfCategoryKey.map((uniqueValueOfCategoryKey, index) => (
+    <Area
+      stroke={PLOT_COLORS[index % PLOT_COLORS.length]}
+      fill={PLOT_SECONDARY_COLORS[index % PLOT_SECONDARY_COLORS.length]}
+      dataKey={uniqueValueOfCategoryKey}
+      type="monotone"
+      strokeWidth={2}
+      name={uniqueValueOfCategoryKey}
+      key={uniqueValueOfCategoryKey}
+      stackId="1"
+      isAnimationActive={false}
+      label={
+        showDataAnnotations ? (
+          <StackedDataAnnotation
+            dataKey={uniqueValueOfCategoryKey}
+            categoryKeyValues={uniqueValuesOfCategoryKey}
+            data={data}
+            dataChangeType={dataAnnotationsChangeType}
+            valueFormatter={yAxisTickFormatter}
+            getCurrentValue={(dataIndex, dataKey) => data[dataIndex][dataKey]}
```

Comment:
This might seem unnecessary at first - `recharts` does pass in a `value` prop to the component we pass in here, but when the chart is stacked, the `value` will be the cumulative, stacked, number, but we want the non-cumulative number. So here I'm introducing a function so that `StackedDataAnnotation` can still report the non-stacked number.